framework/decode/file_processor.cpp

```diff
@@ -1093,28 +1049,10 @@ ParsedBlock BlockParser::ParseMetaData(BlockBuffer& block_buffer)
         success = success && block_buffer.Read(header.memory_offset);
         success = success && block_buffer.Read(header.memory_size);
 
-        if (success)
+        const char*         label       = "fill memory meta-data block";
+        ParameterReadResult read_result = ReadParameterBuffer(label, block_buffer, header.memory_size);
```

Comment:
Do we want to do this if `!success`? If we do, then making the below condition `success && read_result.success` might be slightly more readable.

Reply:
This one got turned inside out accidentally (success check inside result.success check)  fixed.